src/app/machines/views/MachineDetails/MachineNetwork/EditBondForm/EditBondForm.test.tsx

```diff
@@ -394,41 +395,32 @@ describe("EditBondForm", () => {
         system_id: "abc123",
       }),
     ];
+    const link = { id: 1, subnet_id: 1, mode: NetworkLinkMode.AUTO };
     const store = mockStore(state);
-    const wrapper = mount(
-      <Provider store={store}>
-        <MemoryRouter
-          initialEntries={[{ pathname: "/machines", key: "testKey" }]}
-        >
-          <CompatRouter>
-            <EditBondForm
-              close={jest.fn()}
-              nic={bond}
-              selected={[{ nicId: 9 }, { nicId: 10 }]}
-              setSelected={jest.fn()}
-              systemId="abc123"
-            />
-          </CompatRouter>
-        </MemoryRouter>
-      </Provider>
+    renderWithBrowserRouter(
+      <EditBondForm
+        close={jest.fn()}
+        link={link}
+        nic={bond}
+        selected={[{ nicId: 9 }, { nicId: 10 }]}
+        setSelected={jest.fn()}
+        systemId="abc123"
+      />,
+      { route: "/machines", store }
     );
-    submitFormikForm(wrapper, {
-      bond_downdelay: 10,
-      bond_lacp_rate: "fast",
-      bond_mode: BondMode.ACTIVE_BACKUP,
-      bond_miimon: 20,
-      bond_updelay: 30,
-      fabric: 1,
-      ip_address: "1.2.3.4",
-      linkMonitoring: LinkMonitoring.MII,
-      mac_address: "28:21:c6:b9:1b:22",
-      mode: NetworkLinkMode.LINK_UP,
-      name: "bond1",
-      subnet: 1,
-      tags: ["a", "tag"],
-      vlan: 1,
-    });
-    await waitForComponentToPaint(wrapper);
+
+    await userEvent.selectOptions(
+      screen.getByRole("combobox", { name: "Subnet" }),
+      screen.getByRole("option", { name: /test-subnet-1/ })
+    );
+    expect(
+      screen.getByRole("button", { name: "Save interface" })
+    ).toBeEnabled();
+
```

Comment:
```suggestion
```

This assertion is redundant, `userEvent` will throw an error when trying to click an element that is disabled anyway.